### ISSUE TITLE
extern "C" { ... } in headers

### ISSUE THREAD

**Issue by divaykin** [libtrading/libtrading#98]
@penberg @mstanichenko would you be okay to add conditional extern "C" in all the headers?

**Comment by mstanichenko** [comment]
Hi Denis, could you please explain why do we need to put extern "C" in library headers? Isn't it an external application which includes C files should put extern "C" around? Am I missing something?

**Comment by divaykin** [comment]
to simplify library usage in C++ i.e.
```
#ifdef __cplusplus
extern "C" {
#endif

here goes the header file

#ifdef __cplusplus
}
#endif
```

**Comment by mstanichenko** [comment]
Makes sense to me. @penberg, what is the best practice here? Should it be a part of a library or application using it.

**Comment by divaykin** [comment]
see for example https://github.com/libevent/libevent/blob/master/include/event2/buffer.h#L77

**Comment by mstanichenko** [comment]
Yep, as I already said I do not mind at all. @penberg, are you OK with that?

**Comment by penberg** [comment]
@mstanichenko, @divaykin I'm certainly OK with that.

**Comment by penberg** [comment]
Fixed by #103. Closing.
